naga/src/lib.rs

```diff
@@ -2271,4 +2272,10 @@ pub struct Module {
     pub functions: Arena<Function>,
     /// Entry points.
     pub entry_points: Vec<EntryPoint>,
+    /// Arena for all diagnostic filter rules parsed in this module, including those in functions.
+    pub diagnostic_filters: Arena<DiagnosticFilterNode>,
+    /// The head of a linked list of `diagnostic(…)` directives parsed in this module.
```

Comment:
This is probably fine, but I think it's helpful to draw at least a light line between Naga IR concepts and WGSL concepts. So this would be something like:

```rust
/// The leaf node of all module-wide diagnostic filters.
///
/// In WGSL, this includes all global diagnostic filter directives.
```